domain/application/modelmigration/import.go

```diff
@@ -62,14 +64,24 @@ func (i *importOperation) Setup(scope modelmigration.Scope) error {
 
 func (i *importOperation) Execute(ctx context.Context, model description.Model) error {
 	for _, app := range model.Applications() {
-		unitArgs := make([]service.AddUnitParams, 0, len(app.Units()))
+		unitArgs := make([]service.AddUnitArg, 0, len(app.Units()))
 		for _, unit := range app.Units() {
 			name := unit.Name()
-			unitArgs = append(unitArgs, service.AddUnitParams{UnitName: &name})
+			unitArgs = append(unitArgs, service.AddUnitArg{UnitName: &name})
 		}
 
-		err := i.service.CreateApplication(
-			ctx, app.Name(), service.AddApplicationParams{}, unitArgs...,
+		// TODO (stickupkid): This is a temporary solution until we have a
+		// charms in the description model.
+		url, err := internalcharm.ParseURL(app.CharmURL())
+		if err != nil {
+			return fmt.Errorf("parse charm URL %q: %w", app.CharmURL(), err)
+		}
+
+		_, err = i.service.CreateApplication(
+			ctx, app.Name(), &stubCharm{
+				name:     url.Name,
+				revision: url.Revision,
+			}, service.AddApplicationArgs{}, unitArgs...,
```

Comment:
This is a workaround to allow the insertion of a charm to keep RI when adding an application.